# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/gawk/hotfix.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Let's fix linux kernel build:

free(): double free detected in tcache 2
make[4]: *** [arch/x86/Build:9: /usr/src/t2/src.linux.default.20220918.224018.4710.renouvos/linux-5.19/tools/objtool/arch/x86/lib/inat-tables.c] Aborted (core dumped)

--- gawk-5.2.0/interpret.h.vanilla	2022-09-18 23:43:48.161000000 +0200
+++ gawk-5.2.0/interpret.h	2022-09-18 23:44:08.531000000 +0200
@@ -238,6 +238,7 @@
 						_("reference to uninitialized argument `%s'") :
 						_("reference to uninitialized variable `%s'"),
 								save_symbol->vname);
+				UPREF(m);
 				PUSH(m);
 				break;
 
